crates/rolldown/src/module_finalizers/impl_visit_mut.rs

```diff
@@ -34,6 +34,7 @@ impl<'ast> VisitMut<'ast> for ScopeHoistingFinalizer<'_, 'ast> {
       TraverseState::IsTopLevel,
       self.scope_stack.iter().rev().all(|flag| flag.is_block() || flag.is_top()),
     );
+    self.state.set(TraverseState::IsRootLevel, self.scope_stack.iter().all(|flag| flag.is_top()));
```

Comment:
```suggestion
    self.state.set(TraverseState::IsRootLevel, self.scope_stack.iter().rev().all(|flag| flag.is_top()));
```
Maybe this is slightly performant? Also this aligns with the line above.